### PR TITLE
Added two new Trivia Management Routes. Bug fixes and improvements.

### PR DESCRIPTION
**Pull request by Frier03**

### CODE REVIEW

File: src/database/setup.js [hunk 1]
```diff
@@ -6,6 +6,11 @@ const createDatabases = async () => {
         await createDatabase('credentials');
         await createDatabase('trivia_games');
     } catch (error) {
+
+        if (error.code == 'ECONNREFUSED') {
+            console.error('Error: Failed to connect to the MySQL database. Please ensure that the MySQL server is running and accessible.');
```
Comment (Mikkel-Coder) [inline src/database/setup.js]: Perhaps delete 'Error: '

Comment (Mikkel-Coder) [reply src/database/setup.js]: Perhaps also provide the error itself

Comment (Frier03) [reply src/database/setup.js]: > Perhaps delete 'Error: '

Affected file: src/database/setup.js
https://github.com/Syncora/Syncora-Backend/pull/12/commits/d9584972893e8121a3a0de4f30886e943b05cb29

File: src/middleware/authorizationMiddleware.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 const config = require('../config/tokenConfig.json');
```
Comment (Mikkel-Coder) [inline src/middleware/authorizationMiddleware.js]: The names are not consistent

File: src/models/triviaGameModel.js [hunk 3]
```diff
@@ -9,21 +9,69 @@ class TriviaGame {
...
+
+    async save() {
+        const databaseName = process.env.SQL_TRIVIA_GAMES_DB_NAME;
+        const queryStr = 'INSERT INTO game_data (game_id, user_id, game_title, category, questions) VALUES (?, ?, ?, ?, ?)';
+        const values = [this.id, this.author, this.title, this.category, JSON.stringify(this.questions)];
+
+        try {
+            await query(databaseName, queryStr, values);
+
+            console.log('Trivia game saved successfully');
```
Comment (Mikkel-Coder) [inline src/models/triviaGameModel.js]: Perhaps delete?

File: src/routes/triviaRoutes.js [hunk 4]
```diff
@@ -36,9 +31,77 @@ router.post('/', async (req, res) => {
...
+    */
+    const { sub } = req.user;
+    const { ownedGames, limit } = req.query;
+
+    if (!isBooleanString(ownedGames) || !isValidInteger(limit)) {
+        return res.status(400).json({ error: 'Invalid query type parameters.' });
+    }
+
+    try {
+        const options = {
```
Comment (Mikkel-Coder) [inline src/routes/triviaRoutes.js]: Why is this a part of the try? Is it possible not to be a part of the try catch?

File: src/routes/triviaRoutes.js
Comment (Mikkel-Coder) [inline src/routes/triviaRoutes.js:1]: Perhaps check out the standard Google notation for Javascript [link](https://google.github.io/styleguide/jsguide.html)

**Review by Mikkel-Coder — APPROVED** [review]

File: src/middleware/authorizationMiddleware.js (hunk 2 above)
Comment (Frier03) [reply src/middleware/authorizationMiddleware.js]: Solved https://github.com/Syncora/Syncora-Backend/pull/12/commits/002fbcbc8fc658bba2099f0ba8a8a5569b87b741

File: src/models/triviaGameModel.js (hunk 3 above)
Comment (Frier03) [reply src/models/triviaGameModel.js]: Solved. https://github.com/Syncora/Syncora-Backend/pull/12/commits/93fefe0c3400ec385f4349029701c622c497174f

File: src/routes/triviaRoutes.js (hunk 4 above)
Comment (Frier03) [reply src/routes/triviaRoutes.js]: Solved. https://github.com/Syncora/Syncora-Backend/pull/12/commits/dc1f9ebdcae53ff111286490b7f1232e4d8a4c30

File: src/routes/triviaRoutes.js
Comment (Frier03) [reply src/routes/triviaRoutes.js:1]: Won't fix

**Review by Frier03 — COMMENTED** [review]
Looks good for me

**Review by Mikkel-Coder — APPROVED** [review]

**Review by Mikkel-Coder — APPROVED** [review]
Looks go!

**Pull request merged**